recipes-kernel/kernel-tests/kernel-test-nohz.bb

```diff
@@ -31,4 +31,20 @@ do_install_ptest_append() {
     install -m 0755 ${S}/nohz_test ${D}${PTEST_PATH}
 }
 
+pkg_postinst_ontarget_${PN}-ptest_append() {
+    CPUS=`nproc`
+    ISOLATED_CPU=$((CPUS - 1))
+
+    if [ $ISOLATED_CPU -gt 0 ]; then
+        fw_setenv othbootargs 'isolcpus='$ISOLATED_CPU' nohz_full='$ISOLATED_CPU
+    else
+        echo "[error] This test requires a system with 2 or more CPUs"
+        exit 1
+    fi
+}
+
+pkg_postrm_${PN}-ptest_append() {
+    fw_setenv othbootargs
```

Comment:
I'm concerned that this would unset other othbootargs from other ipks as well as this one.

Reply:
That's fair. In practice I don't expect this test to ever get uninstalled (rather the test system gets re-formatted) but let me see if I can do better.

Reply:
Ah right, I forgot this was a ptest.  Then in that case, do better if you want, or simply leave it.  In fact you could even remove the postrm all together.  There's no real harm in leaving othbootargs "dirty" in a test where the system gets formatted.

Reply:
I've added some sed magic in v2 to only add/remove the 'isolcpus' and 'nohz_full' parameters.